server/src/test/java/io/crate/analyze/AlterTableDropColumnAnalyzerTest.java

```diff
@@ -81,6 +81,28 @@ public void test_drop_column_part_of_simple_index() throws Exception {
             dc -> assertThat(dc.ref()).isReference().hasName("b"));
     }
 
+    @Test
+    public void test_drop_column_with_constraint() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE t(a int, b int CHECK (abs(b) + (b / 2) > 2))")
+            .build();
+
+        AnalyzedAlterTableDropColumn d = e.analyze("ALTER TABLE t DROP COLUMN b");
+        assertThat(d.columns()).satisfiesExactly(
+            dc -> assertThat(dc.ref()).isReference().hasName("b"));
+    }
```

Comment:
This test isn't related to the fix or?

Reply:
Not exactly, it's only in Analyzer level, no DropTask code is reached, just to validate that with all other constraints it's valid to drop.